src/functionaltest/java/com/ericsson/ei/frontend/TestSwitchBackend.java

```diff
@@ -0,0 +1,113 @@
+
+import java.io.File;
+import java.io.IOException;
+
+import com.ericsson.ei.frontend.pageobjects.AddBackendPage;
+import com.ericsson.ei.frontend.pageobjects.IndexPage;
+import com.ericsson.ei.frontend.pageobjects.InfoPage;
+import com.ericsson.ei.frontend.pageobjects.SubscriptionPage;
+import com.ericsson.ei.frontend.pageobjects.SwitchBackendPage;
+
+public class TestSwitchBackend extends SeleniumBaseClass{
+    private MockServerClient mockClient1;
+    private MockServerClient mockClient2;
+    private ClientAndServer mockServer1;
+    private ClientAndServer mockServer2;
+
+
+    private static final String BASE_URL = "localhost";
+    private static final int portServer1 = 8090;
+    private static final int portServer2 = 8091;
+
+    private static final String NEW_INSTANCE_SUBSCRIPTION_RESPONSE_FILEPATH = String.join(
+            File.separator, "src", "functionaltest", "resources", "responses", "NewInstanceSubscriptionResponse.json");
+    private static final String DEFAULT_INSTANCE_SUBSCRIPTION_RESPONSE_FILEPATH = String.join(
+            File.separator, "src", "functionaltest", "resources", "responses", "DefaultInstanceSubscriptionResponse.json");
+    private static final String INFORMATION_RESPONSE_FILEPATH = String.join(
+            File.separator, "src", "functionaltest", "resources", "responses", "InformationResponse.json");
+    @Test
+    public void testSwitchBackend() throws Exception {
+        setUpMocks();
+
+        // Open indexpage and verify that it is opened
+        IndexPage indexPageObject = new IndexPage(null, driver, baseUrl);
+        indexPageObject.loadPage();
+        assertEquals("Eiffel Intelligence", indexPageObject.getTitle());
+
+        // Test add backend instance
+        indexPageObject.clickAdminBackendInstancesBtn();
+        AddBackendPage addBackendPage = indexPageObject.clickAddBackendInstanceBtn();
+        SwitchBackendPage switchBackendPage = addBackendPage.addBackendInstance("new_instance", BASE_URL, portServer2, "");
+        assertEquals("new_instance", switchBackendPage.getNewInstanceName());
+
+        // Test switch to the newly added instance
+        switchBackendPage.switchToBackendInstance(1);
+        InfoPage infoPage = indexPageObject.clickEiInfoBtn();
+        assertEquals("http://localhost:" + portServer2, infoPage.getConnectedBackend());
+
+        // Test that different set of subscriptions are available for each instance
+        SubscriptionPage subscriptionPage = indexPageObject.clickSubscriptionPage();
+        assertEquals("new_instance_subscription", subscriptionPage.getSubscriptionNameFromSubscription());
+
+        indexPageObject.clickAdminBackendInstancesBtn();
+        indexPageObject.clickSwitchBackendButton();
+        switchBackendPage.switchToBackendInstance(0);
+        indexPageObject.clickSubscriptionPage();
+        assertEquals("default_instance_subscription", subscriptionPage.getSubscriptionNameFromSubscription());
```

Comment:
I don't agree that it's bad, i think the asserts gives very good failure trace and message. Atleast for those asserts i'm using.

This particular assert gives for 
"example org.junit.ComparisonFailure: expected:<de[]ault_instance_subscr...> but was:<de[f]ault_instance_subscr...>" if i'm giving an incorrect value.

+ stacktrace

Reply:
Look below and you see an assertEquals(false, xxx)
It will state just assertion error expected false got true, there you would want to know what was true.
My comment was on assertEquals as general not just that assertEquals.

Reply:
Ok